runner/diff.go

```diff
@@ -26,6 +26,30 @@ func newIssuesForFile() *issuesForFile {
 	}
 }
 
+// matchFileNameIssueCodes matches if the file name matches an issue code from
+// the analysis result. In case it does, and the check mode is pragma.CheckAll,
+// it changes the check mode to pragma.CheckInclude and adds the matched issue
+// code to the file's issue list.
+func matchFileNameIssueCodes(files map[string]*pragma.File, analysisResult *Result) {
+	fileIssueCodes := make(map[string][]*pragma.File, len(files))
+	for _, file := range files {
+		fileIssueCodes[file.Name] = append(fileIssueCodes[file.Name], file)
+	}
+
+	for _, iss := range analysisResult.Issues {
+		files, ok := fileIssueCodes[iss.Code]
+		if !ok {
+			continue
+		}
+
+		issueCodes := []string{iss.Code}
+		for _, file := range files {
+			file.CheckMode = pragma.CheckInclude
+			file.IssueCodes = issueCodes
+		}
+	}
+}
```

Comment:
Can you flip the logic? `fileIssueCodes` is actually just a map of file names, most of which may not be issue codes.

Instead, making an `analysisIssueCodes` map from analysis results would be better as those are all issue codes. Then check if file name is in that.